---
# Documentation: https://wowchemy.com/docs/managing-content/

title: "Exercises | Spatial Transcriptomics Data Analysis"
subtitle: ""
summary: ""
authors: []
tags: []
categories: []
date: 2020-05-29T15:05:04+01:00
lastmod: 2020-05-29T15:05:04+01:00
featured: false 
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ""
  focal_point: ""
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
---

I recently compiled some exercises focusing on analysis of spatial data for a
course (hosted by SIB). The material spans from very basic analysis
(visualization and clustering) to inference of spatial cell type distributions
by integration of single cell and spatial data. If you are excited about spatial
transcriptomics and what type of analysis you can apply to this type of data,
check out the material
[here](href="https://github.com/fmicompbio/adv_scrnaseq_2020/tree/master/spatial")
. It's all written using standard python libraries for maximal transparency,
allowing you to follow each and every step.

