---
title: 2021 SIB/NBIS course resources
summary: Resources for the lecture "An orientation in the spatial transcriptomics landscape"
date: "2021-07-29T06:13:42Z"

reading_time: false  # Show estimated reading time?
share: true  # Show social sharing links?
profile: true  # Show author profile?
comments: false  # Show comments?

# Optional header image (relative to `assets/media/` folder).
header:
  caption: ""
  image: ""
---
Here you will find links to the publications, resources, tools, and people mentioned in the lecture _"An Orientation in the Spatial Transcriptomics landscape"_ given at the virtual NBIS/SIB course [_2021 Summer School Advanced Topics in Single Cell Analysis_](https://www.sib.swiss/training/course/20210830_ASCTR). The links are sorted according to which slide they relate to, listed in sequential order. The lecture slides are available [here](https://github.com/NBISweden/single-cell_sib_scilifelab_2021/blob/main/lectures/spatial_transcriptomics.pdf).

<hr>

* Slide 1
  * GitHub >> https://github.com/almaan
  * Webpage >> https://almaan.github.io
  * Twitter Handle >> [@aalmaander](https://twitter.com/aalmaander)
* Slide 2
  * Spatial Research Group >> [https://www.spatialresearch.org/](https://www.spatialresearch.org/)
* Slide 3
  * Spatially Resolved Transcriptomes—Next Generation Tools for Tissue Exploration >> DOI [10.1002/bies.201900221](https://onlinelibrary.wiley.com/doi/full/10.1002/bies.201900221)
* Slide 4
  * Method of the year 2020 >> DOI :  [10.1038/s41592-020-01033-y](https://www.nature.com/articles/s41592-020-01033-y)
* Slide 5
  * NA
* Slide 6
  * Spatially Resolved Transcriptomes—Next Generation Tools for Tissue Exploration >> DOI [10.1002/bies.201900221](https://doi.org/10.1002/bies.201900221)
  * Spatially resolved transcriptomics adds a new dimension to genomics >> DOI: [10.1038/s41592-020-01038-7](https://doi.org/10.1038/s41592-020-01038-7)
  * Museum of Spatial Transcriptomics >> https://pachterlab.github.io/LP_2021/
* Slide 7
  * Visium Platform 10x Website >> [LINK](https://www.10xgenomics.com/products/spatial-gene-expression)
* Slide 8
  * NA
* Slide 9
  * Visium specifications >> [link](https://kb.10xgenomics.com/hc/en-us/articles/360035487572-What-is-the-spatial-resolution-and-configuration-of-the-capture-area-of-the-Visium-Gene-Expression-Slide-)
* Slide 10
  * 10x Human Breast cancer data >> [link](https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Breast_Cancer_Block_A_Section_1)
  * 10x Public data sets >> [link](https://support.10xgenomics.com/spatial-gene-expression/datasets)
* Slide 11
  * NA
* Slide 12
  * Seurat >> [link](https://satijalab.org/seurat/)
  * Scanpy >> [link](https://scanpy.readthedocs.io/en/stable/)
  * STUtility >> [link](https://ludvigla.github.io/STUtility_web_site/)
  * stLearn >> [link](https://github.com/BiomedicalMachineLearning/stLearn)
  * SpatialExperiment >> [link](https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html)
  * Giotto >> [link](http://spatialgiotto.rc.fas.harvard.edu/)
* Slide 13
  * Spatial Deconvolution of HER2-positive Breast Tumors Reveals Novel Intercellular Relationships >> DOI : [10.1101/2020.07.14.200600](https://www.biorxiv.org/content/10.1101/2020.07.14.200600v1)
* Slide 14
  * NA
* Slide 15
  * NA
* Slide 16
  * NA
* Slide 17
  * NA
* Slide 18
  * NA
* Slide 19
  * Single-cell and spatial transcriptomics enables probabilistic inference of cell type topography >> DOI : [10.1038/s42003-020-01247-y](https://www.nature.com/articles/s42003-020-01247-y)
* Slide 20
  * Moncada et al. >> DOI : [10.1038/s41587-019-0392-8](https://www.nature.com/articles/s41587-019-0392-8)
  * Seurat spatial and single cell data integration >> [link](https://satijalab.org/seurat/archive/v3.2/spatial_vignette.html)
  * stereoscope GitHub >> [https://github.com/almaan/stereoscope](https://github.com/almaan/stereoscope)
  * stereoscope paper >> DOI : [10.1038/s42003-020-01247-y](https://www.nature.com/articles/s42003-020-01247-y)
  * RCTD GitHub >> [https://github.com/dmcable/RCTD/](https://github.com/dmcable/RCTD/)
  * RCTD paper >> DOI : [10.1038/s41587-021-00830-w](https://www.nature.com/articles/s41587-021-00830-w)
  * cell2location GitHub >> [https://github.com/BayraktarLab/cell2location](https://github.com/BayraktarLab/cell2location)
  * cell2location preprint >> DOI : [10.1101/2020.11.15.378125](https://doi.org/10.1101/2020.11.15.378125 )
  * Tangram GitHub >> https://github.com/broadinstitute/Tangram
  * Tangram preprint >> DOI : [10.1101/2020.08.29.272831](https://www.biorxiv.org/content/10.1101/2020.08.29.272831v1) 
* Slide 20
  * stereoscope GitHub >> https://github.com/almaan/stereoscope
  * stereoscope paper >> DOI : [10.1038/s42003-020-01247-y](https://doi.org/10.1038/s42003-020-01247-y)
* Slide 21
  * See Slide 20
  * NMFReg tutorial >> DOI : [https://github.com/tudaga/NMFreg_tutorial](https://github.com/tudaga/NMFreg_tutorial)
  * SPOTlight paper >> DOI : [https://doi.org/10.1093](https://doi.org/10.1093)
  * spatialDWLS paper >> DOI : [https://doi.org/10.1186/s13059-021-02362-7](https://doi.org/10.1186/s13059-021-02362-7)
* Slide 22
  * stereoscope paper >> DOI : [10.1038/s42003-020-01247-y](https://www.nature.com/articles/s42003-020-01247-y)
  * stereoscope GitHub >> [https://github.com/almaan/stereoscope](https://github.com/almaan/stereoscope)
* Slide 23
  * Developmental heart data source >> DOI: [10.1016/j.cell.2019.11.025](10.1016/j.cell.2019.11.025)
  * 10x Mouse Brain Visium data >> [link](https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain)
  * Mouse Brain Single Cell data >> [link](http://loom.linnarssonlab.org/dataset/cellmetadata/Mousebrain.org.level1/L1_Hippocampus.loom)
* Slide 24
  * stereoscope supplementary >> [link](https://static-content.springer.com/esm/art%3A10.1038%2Fs42003-020-01247-y/MediaObjects/42003_2020_1247_MOESM1_ESM.pdf)
* Slide 25
  * Robust decomposition of cell type mixtures in spatial transcriptomics >> DOI : [10.1038/s41587-021-00830-w](https://www.nature.com/articles/s41587-021-00830-w)
* Slide 26
  * Deep learning and alignment of spatially-resolved whole transcriptomes of single cells in the mouse brain with Tangram >> DOI : [10.1101/2020.08.29.272831](https://www.biorxiv.org/content/10.1101/2020.08.29.272831v1)
* Slide 27
  * NA
* Slide 28
  * Open Problems in Single-Cell Analysis >> [https://openproblems.bio/](https://openproblems.bio/)
  * Spatial Decomposition task issue >> [link](https://github.com/openproblems-bio/openproblems/issues/233)
* Slide 29
  * NA
* Slide 30
  * SpatialDE GitHub >> [https://github.com/Teichlab/SpatialDE](https://github.com/Teichlab/SpatialDE)
  * SpatialDE paper >> DOI : [doi.org/10.1038/nmeth.4636](https://www.nature.com/articles/nmeth.4636)
  * SVCA GitHub >> [https://github.com/damienArnol/svca](https://github.com/damienArnol/svca)
  * SVCA paper >> DOI : [10.1016/j.celrep.2019.08.077](https://www.sciencedirect.com/science/article/pii/S2211124719311325)
  * SPARK GitHub >> [https://github.com/xzhoulab/SPARK](https://github.com/xzhoulab/SPARK)
  * SPARK paper >> DOI : [10.1038/s41592-019-0701-7](https://www.nature.com/articles/s41592-019-0701-7)
  * Gaussian Processes Info >> [link](https://distill.pub/2019/visual-exploration-gaussian-processes/)
* Slide 31
  * Identification of spatially associated subpopulations by combining scRNAseq and sequential 
* Slide 32
  * NA
* Slide 33
  * NA 
* Slide 34
 * Identification of spatially associated subpopulations by combining scRNAseq and sequential fluorescence in situ hybridization data >> DOI : [https://doi.org/10.1038/nbt.4260](https://doi.org/10.1038/nbt.4260)
* Slide 35
  * STARCH publication >> DOI : [10.1088/1478-3975/abbe99 ](https://iopscience.iop.org/article/10.1088/1478-3975/abbe99)
  * scHOT publication >> DOI : [10.1038/s41592-020-0885-x](https://www.nature.com/articles/s41592-020-0885-x)
* Slide 36
  * MULTILAYER publication >> DOI : [10.1016/j.cels.2021.04.008](https://www.sciencedirect.com/science/article/pii/S2405471221001514)
* Slide 37
  * Science publication >> DOI : [10.1126/science.aav9776](https://science.sciencemag.org/content/364/6435/89)
  * Splotch publication >> DOI : [10.1101/757096](https://www.biorxiv.org/content/10.1101/757096v1)
* Slide 38
  * NA
Slide 39
  * NA
* Slide 40
  * XFuse GitHub >> [https://github.com/ludvb/xfuse](https://github.com/ludvb/xfuse)
  * stPlus GitHub >> [https://github.com/xy-chen16/stPlus](https://github.com/xy-chen16/stPlus)
  * SpaGCN GitHub >> [https://github.com/jianhuupenn/SpaGCN](https://github.com/jianhuupenn/SpaGCN)
  * RESEPT GitHub >> [https://github.com/OSU-BMBL/RESEPT](https://github.com/OSU-BMBL/RESEPT)
* Slide 41
  * Super-resolved spatial transcriptomics by deep data fusion (prepint) >> DOI : [10.1101/2020.02.28.963413 ](https://www.biorxiv.org/content/10.1101/2020.02.28.963413v1)
  * XFuse GitHub >> [https://github.com/ludvb/xfuse](https://github.com/ludvb/xfuse)
* Slide 42
  * Squidpy >> [https://squidpy.readthedocs.io/](https://squidpy.readthedocs.io/)
* Slide 43
  * NA
* Slide 44
  * HER2 study code >> [https://github.com/almaan/her2st/](https://github.com/almaan/her2st/)
  * HER2 study paper >> DOI : [10.1101/2020.07.14.200600](https://www.biorxiv.org/content/10.1101/2020.07.14.200600v1)
* Slide 45
  * Swarbrick Lab >> [link](https://www.garvan.org.au/research/cancer/tumour-progression)
  * Human breast cancer single cell data >> DOI : [10.5281/zenodo.4739739](https://doi.org/10.5281/zenodo.4739739)
* Slide 46
  * NA
* Slide 47
  * NA
* Slide 48
  * Camilla Engblom Twitter Handle >> [@camillaengblom](https://twitter.com/camillaengblom)
  * Camilla Engblom Publications >> [Google Scholar](https://scholar.google.com/citations?user=lXDdWjAAAAAJ&hl=en)
* Slide 49
  * NA
* Slide 50
  * NA
* Slide 51
  * NA
* Slide 52
  * HER2 study code >> [https://github.com/almaan/her2st/](https://github.com/almaan/her2st/)
  * HER2 study paper >> DOI : [10.1101/2020.07.14.200600](https://www.biorxiv.org/content/10.1101/2020.07.14.200600v1)
* Slide 53
  * Liver study code >> [https://github.com/almaan/ST-mLiver](https://github.com/almaan/ST-mLiver)
  * Liver study paper >> DOI [10.1101/2021.01.11.426100](https://doi.org/10.1101/2021.01.11.426100)
* Slide 54
  * NA
* Slide 55
  * Liver study code >> [https://github.com/almaan/ST-mLiver](https://github.com/almaan/ST-mLiver)
  * Liver study paper >> DOI [10.1101/2021.01.11.426100](https://doi.org/10.1101/2021.01.11.426100)
* Slide 56
  * Liver study code >> [https://github.com/almaan/ST-mLiver](https://github.com/almaan/ST-mLiver)
  * Liver study paper >> DOI [10.1101/2021.01.11.426100](https://doi.org/10.1101/2021.01.11.426100)
* Slide 57
  * NA
* Slide 58
  * Spatial Research Group >> [www.spatialresearch.org](www.spatialresearch.org)
  * 10x >> [https://www.10xgenomics.com/](https://www.10xgenomics.com/)
  * EMBL >> [https://www.embl.org](https://www.embl.org)
  * Tommaso Biancalani >> [Google Scholar](https://scholar.google.it/citations?user=s_qd9x0AAAAJ&hl=en)
  * Dyalan Cable >> [Google Scholar](https://scholar.google.com/citations?user=5XSOqjwAAAAJ&hl=en)
* Slide 60
  * GitHub >> https://github.com/almaan
  * Webpage >> https://almaan.github.io
  * Twitter Handle >> [@aalmaander](https://twitter.com/aalmaander)







